x-pack/plugins/security_solution/server/endpoint/services/metadata/endpoint_metadata_service.ts

```diff
@@ -0,0 +1,238 @@
+    let fleetAgent: Agent | undefined;
+
+    // Get Fleet agent
+    try {
+      if (!fleetAgentId) {
+        fleetAgentId = endpointMetadata.agent.id;
+        this.logger?.warn(`Missing elastic agent id, using host id instead ${fleetAgentId}`);
+      }
+
+      fleetAgent = await this.getFleetAgent(esClient, fleetAgentId);
+    } catch (error) {
+      if (error instanceof FleetAgentNotFoundError) {
+        this.logger?.warn(`agent with id ${fleetAgentId} not found`);
+      } else {
+        throw error;
+      }
+    }
+
+    // If the agent is not longer active, then that means that the Agent/Endpoint have been un-enrolled from the host
+    if (fleetAgent && !fleetAgent.active) {
+      throw new EndpointHostUnEnrolledError(
+        `Endpoint with id ${endpointId} (Fleet agent id ${fleetAgentId}) is unenrolled`
+      );
+    }
+
+    // ------------------------------------------------------------------------------
+    // Any failures in enriching the Host form this point should NOT cause an error
+    // ------------------------------------------------------------------------------
+    try {
+      let fleetAgentPolicy: AgentPolicyWithPackagePolicies | undefined;
+      let endpointPackagePolicy: PackagePolicy | undefined;
+
+      // Get Agent Policy and Endpoint Package Policy
+      if (fleetAgent) {
+        try {
+          fleetAgentPolicy = await this.getFleetAgentPolicy(fleetAgent.policy_id!);
+          endpointPackagePolicy = fleetAgentPolicy.package_policies.find(
+            (policy) => policy.package?.name === 'endpoint'
+          );
+        } catch (error) {
+          this.logger?.error(error);
+        }
+      }
+
+      return {
+        metadata: endpointMetadata,
+        host_status: fleetAgent
+          ? fleetAgentStatusToEndpointHostStatus(this.agentService.getStatusForAgent(fleetAgent))
+          : DEFAULT_ENDPOINT_HOST_STATUS,
+        policy_info: {
+          agent: {
+            applied: {
+              revision: fleetAgent?.policy_revision ?? 0,
+              id: fleetAgent?.policy_id ?? '',
+            },
+            configured: {
+              revision: fleetAgentPolicy?.revision ?? 0,
+              id: fleetAgentPolicy?.id ?? '',
+            },
+          },
+          endpoint: {
+            revision: endpointPackagePolicy?.revision ?? 0,
+            id: endpointPackagePolicy?.id ?? '',
+          },
+        },
+      };
+    } catch (error) {
+      throw wrapErrorIfNeeded(error);
+    }
+  }
+
+  /**
+   * Retrieve a single Fleet Agent data
+   *
+   * @param esClient Elasticsearch Client (usually scoped to the user's context)
+   * @param agentId The elastic agent id (`from `elastic.agent.id`)
+   */
+  async getFleetAgent(esClient: ElasticsearchClient, agentId: string): Promise<Agent> {
+    try {
+      const agent = await this.agentService.getAgent(esClient, agentId);
+      agent.status = this.agentService.getStatusForAgent(agent);
```

Comment:
I am wondering if it will make sense to add the status to the `agentService.getAgent` method it will be easier to consume

Reply:
@nchaulet  - Yeah, I was wondering the same thing. Looks like the status is currently added in the route handler after calling the service method (`getAgent()`). I can make the change if you are ok with it. If we do that, then I don't think we need to expose this new `getStatusForAgent()` method.

Reply:
Yes let's do the change it will make everything easier, do you mind doing the change for `getAgents` too for consistency?

Reply:
Not at all. I'll make the changes 👍

Reply:
@nchaulet done. Take another look when you get a chance. The change was done on the helper (mapper) utility rather than individual service functions, so they should all be covered